src/libostree/ostree-sysroot.c

```diff
@@ -826,7 +838,7 @@ parse_deployment (OstreeSysroot *self, const char *boot_link, OstreeDeployment *
        * be relied on anyways.
        */
       is_booted_deployment
-          = (stbuf.st_dev == self->root_device && stbuf.st_ino == self->root_inode)
+          = (stbuf.st_dev == expected_root_dev && stbuf.st_ino == expected_root_inode)
             || (etc_stbuf.st_dev == self->etc_device && etc_stbuf.st_ino == self->etc_inode);
```

Comment:
We should also be able to drop this conditional too, let's try to do that after some more testing.